core/src/main/java/org/frankframework/jdbc/JdbcQuerySenderBase.java

```diff
@@ -358,34 +357,31 @@ protected PipeRunResult executeStatementSet(@Nonnull QueryExecutionContext query
 					throw new IllegalStateException("Unsupported queryType: ["+queryExecutionContext.getQueryType()+"]");
 			}
 		} catch (SenderException e) {
-			if (e.getCause() instanceof SQLException sqle) {
-				if  (sqle.getErrorCode() == 1013) {
-					throw new TimeoutException("Timeout of ["+getTimeout()+"] sec expired");
-				}
+			if (e.getCause() instanceof SQLException sqle && sqle.getErrorCode() == 1013) {
+				throw new TimeoutException("Timeout of ["+getTimeout()+"] sec expired");
 			}
 			throw new SenderException(e);
 		} catch (Throwable t) {
 			throw new SenderException(getLogPrefix() + "got exception sending message", t);
 		} finally {
 			closeStatementSet(queryExecutionContext);
 			ParameterList newParameterList = queryExecutionContext.getParameterList();
-			if (isCloseInputstreamOnExit() && newParameterList!=null) {
-				for (int i = 0; i < newParameterList.size(); i++) {
-					IParameter param = newParameterList.getParameter(i);
-					if (param.getType() == ParameterType.INPUTSTREAM) {
-						log.debug("{}Closing inputstream for parameter [{}]", this::getLogPrefix, param::getName);
-						try {Object object = newParameterList.getParameter(i).getValue(null, message, session, true);
-							if(object instanceof AutoCloseable closeable) {
-								closeable.close();
-							}
-							else {
-								log.error("unable to auto-close parameter [{}]", param.getName());
+			if (isCloseInputstreamOnExit() && newParameterList != null) {
+				newParameterList.stream()
+						.filter(param -> param.getType() == ParameterType.INPUTSTREAM)
+						.forEach(param -> {
+							log.debug("{}Closing inputstream for parameter [{}]", this::getLogPrefix, param::getName);
+							try {
+								Object object = param.getValue(null, message, session, true);
+								if (object instanceof AutoCloseable closeable) {
+									closeable.close();
+								} else {
+									log.error("{}unable to auto-close parameter [{}]", this::getLogPrefix, param::getName);
+								}
+							} catch (Exception e) {
+								log.warn(new SenderException(getLogPrefix() + "got exception closing inputstream", e));
 							}
-						} catch (Exception e) {
-							log.warn(new SenderException(getLogPrefix() + "got exception closing inputstream", e));
-						}
-					}
-				}
+						});
```

Comment:
Kwestie van stijl, maar als ik een body nodig heb in een lambda, schrijf ik dat meestal uit in een aparte method. Bijvoorbeeld:

`.forEach(param -> closeInputStreamForParameter(param, message, session));`

en dan:

```
	private void closeInputStreamForParameter(IParameter param, Message message, PipeLineSession session) {
		log.debug("{}Closing inputstream for parameter [{}]", this::getLogPrefix, param::getName);
		try {
			Object object = param.getValue(null, message, session, true);
			if (object instanceof AutoCloseable closeable) {
				closeable.close();
			} else {
				log.error("{}unable to auto-close parameter [{}]", this::getLogPrefix, param::getName);
			}
		} catch (Exception e) {
			log.warn(new SenderException(getLogPrefix() + "got exception closing inputstream", e));
		}
	}
```

Reply:
Ja dat doe ik normaal ook, klopt. Maakt het veel leesbaarder. Ik had deze keer gekozen voor automatische omzetting naar Stream operation door IntelliJ en er toen niet meer aan gedaan.
Ik kijk er nog even naar.